llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -8308,6 +8367,115 @@ llvm::Function *SPIRVToLLVM::createLibraryEntryFunc() {
   return func;
 }
 
+const Vkgc::PipelineOptions *SPIRVToLLVM::getPipelineOptions() const {
+  assert(m_context && "Invalid context!");
+
+  return static_cast<Llpc::Context *>(m_context)->getPipelineContext()->getPipelineOptions();
+}
+
+template <> SPIRVValue *SPIRVToLLVM::getValueForScratchBoundaryCheckWrapper<SPIRVStore>(SPIRVStore *toBeWrapped) const {
+  return toBeWrapped->getDst();
+}
+
+template <> SPIRVValue *SPIRVToLLVM::getValueForScratchBoundaryCheckWrapper<SPIRVLoad>(SPIRVLoad *toBeWrapped) const {
+  return toBeWrapped->getSrc();
+}
+
+template <class T>
+std::pair<bool, SPIRVToLLVM::ScratchBoundaryCheckInfo> SPIRVToLLVM::prepareScratchBoundaryCheck(T *origin) {
+  auto const nullValue = std::make_pair<bool, ScratchBoundaryCheckInfo>(false, {});
```

Comment:
nit: don't use `make_pair` with explicit type arguments, `std::pair<...>(...)` should work fine. Given that this function is used in multiple places, consider using a new struct as the return type instead. This would make it clearer what the `bool` is for.